eth/watchers/senderwatcher_test.go

```diff
@@ -708,3 +708,54 @@ func TestNewRoundEvent_LogRemoved(t *testing.T) {
 	assert.Nil(err)
 	assert.Equal(claimed, expectedClaimedAmount)
 }
+
+func TestSubscribeSenderInfoChange(t *testing.T) {
+	assert := assert.New(t)
+	require := require.New(t)
+	watcher := &stubBlockWatcher{}
+	lpEth := &eth.StubClient{
+		SenderInfo: &pm.SenderInfo{
+			Reserve: &pm.ReserveInfo{
+				FundsRemaining: big.NewInt(100),
+			},
+		},
+	}
+
+	sm, err := NewSenderWatcher(stubTicketBrokerAddr, watcher, lpEth, &stubTimeWatcher{})
+	require.Nil(err)
+
+	header := defaultMiniHeader()
+
+	// Test reserve funded
+	fundedEvent := newStubReserveFundedLog()
+	header.Logs = append(header.Logs, fundedEvent)
+
+	blockEvent := &blockwatch.Event{
+		Type:        blockwatch.Added,
+		BlockHeader: header,
+	}
+
+	sink := make(chan ethcommon.Address, 10)
+	sub := sm.SubscribeSenderInfoChange(sink)
+	defer sub.Unsubscribe()
+
+	go sm.Watch()
+	defer sm.Stop()
+	time.Sleep(time.Millisecond)
+
+	// set initial values
+	_, err = sm.GetSenderInfo(stubSender)
+	require.Nil(err)
+
+	watcher.sink <- []*blockwatch.Event{blockEvent}
```

Comment:
I suspect that the time.Sleep calls here can cause this test to be flaky when run in CI. Could we write this test without using time.Sleep by directly calling the `handleLog()` method?

Reply:
No because that would be a cyclical import

(`eth/watchers` imports `pm` and the proposed change would require to import `eth/watchers` in `pm`

Furthermore the `handleLog` method is not exported from `SenderWatcher` anyway and we should not contaminate the API if it's not necessary for external callers other than unit tests I think.

Reply:
Not sure how a cyclical import is an issue here. Just to make sure we're on the same page:

- I am not suggesting to make any changes to the `SenderWatcher` API
- I am suggesting to *only* make changes in this particular test
- I am suggesting to call `SenderWatcher.handleLog()` to trigger a send into `SenderWatcher.senderInfoFeed`. This should accomplish the same thing as the current code in the test that sends into `watcher.sink`. The difference is that we would not need to deal with waiting (via time.Sleep) for the `SenderWatcher` watch loop to process the value sent into `watcher.sink`

Reply:
Oh sorry I was looking at the relevant tests in `sendermonitor_test` instead.

Reply:
While I included the change for now I feel like this change request might be a little premature since the test has not proven to be flaky and none of the tests in that file call that method directly and have worked just fine.

Reply:
Ah after taking a second look at that test I realized that the assertions are placed after a statement that blocks until a value is received on a channel (i.e. `sender := <-sink`). Since the assertion is not run until after a value is received on a channel flaky test behavior from waiting for goroutines shouldn't be a problem. 

The commit that addressed the original commit is still useful though as it more directly triggers the desired state update (in `handleLog()`) which helps test readability.